pageserver/src/http/openapi_spec.yml

```diff
@@ -363,11 +363,29 @@ paths:
         * MUST NOT ASSUME that the request has been lost, based on the observation
           that a subsequent tenant status request returns 404. The request may
           still be in flight. It must be retried.
+
+        The client SHOULD supply a `TenantConfig` for the tenant in the request body.
+        Settings specified in the config override the pageserver's defaults.
+        It is guaranteed that the config settings are applied before the pageserver
+        starts operating on the tenant. E.g., if the config specifies a specific
+        PITR interval for a tenant, then that setting will be in effect before the
+        pageserver starts the garbage collection loop. This enables a client to
+        guarantee a specific PITR setting across detach/attach cycles.
+        The pageserver will reject the request if it cannot parse the config, or
+        if there are any unknown fields in it.
+
+        If the client does not supply a config, the pageserver will use its defaults.
+        This behavior is deprecated: https://github.com/neondatabase/neon/issues/4282
```

Comment:
So do you imagine this as we always specify **all** settings in the call? This will make adding new settings very painful, as first you need to rollout new control-plane that will start sending new options and only then rollout new storage that will start accepting them (old will just drop unknown attributes, iiuc)

I'd still consider that we use everything by default if there is no override from control-plane

Reply:
> So do you imagine this as we always specify all settings in the call? 

No. The minimum acceptable body is either the empty body (for backwards compat, deprectated, see https://github.com/neondatabase/neon/issues/4282 ) or `{ "config": {} }`